Lib/ssl.py

```diff
@@ -327,12 +327,20 @@ def _inet_paton(ipname):
     Supports IPv4 addresses on all platforms and IPv6 on platforms with IPv6
     support.
     """
-    # inet_aton() also accepts strings like '1'
-    if ipname.count('.') == 3:
-        try:
-            return _socket.inet_aton(ipname)
-        except OSError:
-            pass
+    # inet_aton() also accepts strings like '1', '127.1', or
+    # '127.0.0.1 whatever'.
+    try:
+        addr = _socket.inet_aton(ipname)
+    except OSError:
+        pass
+    else:
+        if _socket.inet_ntoa(addr) == ipname:
+            # only accept injective ipnames
+            return addr
+        else:
+            raise ValueError(
+                "{!r} is not a quad-dotted IPv4 address.".format(ipname)
```

Comment:
FWIW, on Windows if you pass an invalid address you get `OSError: illegal IP address string passed to inet_aton`

Reply:
Some implementations of ``inet_aton`` are strict and don't allow trailing data. glibc's ``inet_aton`` is more lean and ignores trailing data if the first character is some sort of white space.